File: lib/model/hargamitra.dart
import 'jenisbarang.dart';
class HargaMitra {
  final int idHargaMitra;
  final int hargaSewa;
  final JenisBarang? jenisBarang;

  HargaMitra({
    required this.idHargaMitra,
    required this.hargaSewa,
    this.jenisBarang,
  });

  factory HargaMitra.fromJson(Map<String, dynamic> json) {
    return HargaMitra(
      idHargaMitra: json['harga_mitra'],
      hargaSewa: json['harga_sewa'],
      jenisBarang: json['jenis_barang'] != null 
          ? JenisBarang.fromJson(json['jenis_barang']) 
          : null,
    );
  }
}

File: app/Models/HargaMitra.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class HargaMitra extends Model
{
    protected $table='harga_mitra';
    protected $primaryKey = 'id_harga_mitra';
    public $timestamps = false;
    protected $fillable=[
        'id_mitra',
        'id_jenis_barang',
        'harga_sewa'
    ];
    public function mitra(){
        return $this->belongsTo(Mitra::class,'id_mitra');
    }
    public function jenis_barang(){
        return $this->belongsTo(JenisBarang::class,'id_jenis_barang');}
}
